### DISCUSSION TITLE
Performance is really slow "running file create/rename/delete options"

### DISCUSSION THREAD

**Discussion by ghost** [coder/code-server#2923] [Q&A]
In some of the recent build tags on docker hub I am seeing really bad performance, moving around and renaming files throw up a VS Code message like **running file create, rename, move, delete operations**.

Anyone have clues what might be up? Don't want to make a bug report if it's not reproducible.

**Comment by JammSpread** [comment]
I agree, compared to deleting a file using a command like, `rm`, it indeed it quite slow. I think this may be due to the methods VS Code uses to complete these operations.

**Reply by oxy** [reply]
Hi!

I'm not aware of any changes that should impact this, and in my experience its been just fine - do you know when the performance regression happened?

**Reply by JammSpread** [reply]
This is the chain of processes that I've been able to find for at least deleting a file permanently.
This should be pretty similar to the other operations.
- https://github.com/cdr/code-server/blob/d8e92255065bc8edfa95a26f178fef9685c96a44/lib/vscode/src/vs/workbench/contrib/files/browser/fileActions.contribution.ts#L97
- https://github.com/cdr/code-server/blob/d8e92255065bc8edfa95a26f178fef9685c96a44/lib/vscode/src/vs/workbench/contrib/files/browser/fileActions.ts#L943
- https://github.com/cdr/code-server/blob/d8e92255065bc8edfa95a26f178fef9685c96a44/lib/vscode/src/vs/workbench/contrib/files/browser/explorerService.ts#L151
- https://github.com/cdr/code-server/blob/d8e92255065bc8edfa95a26f178fef9685c96a44/lib/vscode/src/vs/platform/files/node/diskFileSystemProvider.ts#L407
- https://github.com/cdr/code-server/blob/d8e92255065bc8edfa95a26f178fef9685c96a44/lib/vscode/src/vs/platform/files/node/diskFileSystemProvider.ts#L419
- https://github.com/cdr/code-server/blob/d8e92255065bc8edfa95a26f178fef9685c96a44/lib/vscode/src/vs/base/node/pfs.ts#L81
Then I think it uses `fs.unlink`.
So underlying it uses `fs` with a promisify.

**Reply by JammSpread** [reply]
I think it being asynchronous might slow down the process.

**Reply by JammSpread** [reply]
You can test the performance by generating a 1GB (something less might work as well) file.
You can generate one using the `dd` command on Linux:
```bash
dd of=file.txt bs=1G count=1
```

**Reply by pjknkda** [reply]
Possibly related issue is resolved in the upstream (https://github.com/microsoft/vscode/pull/144890) and is released with vscode 1.66.0. Please check 😄
